installer/installer.go

```diff
@@ -269,19 +270,24 @@ func (s *Stack) createKeyPair() error {
 		keypairName = s.SSHKeyName
 	}
 	if keypair, err := loadSSHKey(keypairName); err == nil {
-		s.SSHKey = keypair
-		s.SSHKeyName = keypairName
-		res, err := s.ec2.DescribeKeyPairs(&ec2.DescribeKeyPairsRequest{
-			Filters: []ec2.Filter{{
-				Name:   aws.String("key-name"),
-				Values: []string{keypairName},
-			}},
-		})
-		if err == nil && len(res.KeyPairs) > 0 && *res.KeyPairs[0].KeyName == keypairName {
-			// key exists, we're good to go
-			// TODO(jvatic): verify key fingerprint
-			s.SendEvent(fmt.Sprintf("Using saved key pair (%s)", keypairName))
-			return nil
+		if fingerprint, err := awsutil.FingerprintImportedKey(keypair.PrivateKey); err == nil {
+			s.SSHKey = keypair
+			s.SSHKeyName = keypairName
+			res, err := s.ec2.DescribeKeyPairs(&ec2.DescribeKeyPairsRequest{
+				Filters: []ec2.Filter{{
+					Name:   aws.String("key-name"),
+					Values: []string{keypairName},
+				},
+					{
+						Name:   aws.String("fingerprint"),
+						Values: []string{fingerprint},
+					}},
+			})
+			if err == nil && len(res.KeyPairs) > 0 && *res.KeyPairs[0].KeyName == keypairName {
```

Comment:
I don't really like these nested `err == nil` checks. Is there a way to invert the logic or split it into separate functions?